parsers/ESIOS.py

```diff
@@ -45,7 +45,19 @@ def fetch_exchange(
     end_date = target_datetime.isoformat()
     dates = {"start_date": start_date, "end_date": end_date}
     query = urlencode(dates)
-    url = "https://api.esios.ree.es/indicators/10209?{0}".format(query)
+
+    zone_key = ZoneKey("->".join(sorted([zone_key1, zone_key2])))
+    if zone_key == ZoneKey("AD->ES"):
+        url = "https://api.esios.ree.es/indicators/10278?{0}".format(query)
+    elif zone_key == ZoneKey("ES->MA"):
+        url = "https://api.esios.ree.es/indicators/10209?{0}".format(query)
+    else:
```

Comment:
While this works well for this change I had another approach in mind to make this more extensible and easier to re-use in the future. (for example adding demand parsers for the Spanish islands).

If you create a format_url function (or get_url might be a better name) like this:
```python
def format_url(target_datetime: datetime, ID: str):
    start_date = (target_datetime - timedelta(hours=24)).isoformat()
    end_date = target_datetime.isoformat()
    dates = {"start_date": start_date, "end_date": end_date}
    query = urlencode(dates)
    return f"https://api.esios.ree.es/indicators/{ID}?{query}"
```
And then create a mapping like this (as a constant above the other code):
```python
EXCHANGE_ID_MAP = {
    "AD->ES": "10210",
    "ES->MA": "10209",
}
```
We can call it like this:
```python
    url = format_url(target_datetime, EXCHANGE_ID_MAP[zone_key])
```

This means we can reuse the format_url function and just switch out the mapping if we create new parser functions.